File: LibraryMe.API/BookLibrary/Models/DTO/BookDTO.cs
using BookLibrary.Models.Domain;

namespace BookLibrary.Models.DTO
{
    public class BookDTO
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string ImageUrl { get; set; }

        public IEnumerable<GenreDTO> Genres { get; set; }
        public IEnumerable<AuthorLinkDTO> Authors { get; set; }
    }
}

File: LibraryMe.UI/LibraryMe/src/app/Features/Books/models/book-dto.ts
import {GenreDTO} from "../Genres/models/genre-dto";
import {AuthorLinkDTO} from "../../Authors/models/author-link-dto";

export interface BookDTO {
    id:string;
    totalAmount: number;
    availableAmount: number;
    title: string;
    description: string;
    imageUrl: string;
    genres: GenreDTO[];
    authors: AuthorLinkDTO[];
}
